src/shims/sync.rs

```diff
@@ -55,75 +77,89 @@ fn mutexattr_set_kind<'mir, 'tcx: 'mir>(
 // bytes 0-3: reserved for signature on macOS
 // (need to avoid this because it is set by static initializer macros)
 // bytes 4-7: count of how many times this mutex has been locked, as a u32
+// bytes 8-11: when count > 0, id of the owner thread as a u32
 // bytes 12-15 or 16-19 (depending on platform): mutex kind, as an i32
 // (the kind has to be at its offset for compatibility with static initializer macros)
+// bytes 20-23: when count > 0, id of the blockset in which the blocked threads are waiting.
```

Comment:
The blockset is reused, right? It doesn't get reset when `count == 0`?
Maybe better say "blockset on which blocked threads are waiting (or 0 if blockset is not yet assigned)", or so.

Same for RwLock.